pkg/telemetry/telemetryservice.go

```diff
@@ -82,15 +83,15 @@ type TelemetryService interface {
 }
 
 const (
-	workerCleanupWait  = 3 * time.Minute
-	jobQueueBufferSize = 10000
+	workerCleanupWait = 3 * time.Minute
+	jobsQueueMinSize  = 2048
 )
 
 type telemetryService struct {
 	AnalyticsService
 
-	notifier webhook.QueuedNotifier
-	jobsChan chan func()
+	notifier  webhook.QueuedNotifier
+	jobsQueue *utils.OpsQueue
```

Comment:
in the new queue implementation events that haven't been processed when `Stop` is called are dropped. this probably isn't desirable for telemetry. we might need separate `Stop` and `StopNow` functions

Reply:
Good point. Will check on that.